### ISSUE TITLE
save colors into some kind of color panel for easy access

### ISSUE THREAD

**Issue by jesse-rb** [jesse-rb/qolboard-spa#23]
Either automatically save recently used colors, or add ability to quickly save the current color (I think having both in two panels next to each other would be ideal). For the latter, the saved colors might need to be able to be edited as well (e.g. using html color inputs to show these would allow them to be edited pretty easily too!

I think a good strategy to build this could be a new svelte component that we would use anywhere we would normally/are currently using an html color input. This component can consist of the main color picker (can be as simple as the normal html color input, but would need some kinds of panel/options underneath or around to save/have quick access to select recent/saved colors. This component could possibly make use of it's own [svelte store](https://learn.svelte.dev/tutorial/writable-stores) to hold consistent saved colors everywhere the component is used.

Once the functionality is done, the component state would eventually need to be added to local storage along with other canvas related state (see [example component serialize/deerialize functions](https://github.com/jesse-rb/qolboard-spa/blob/a86e81c9e128c23ab87dcc2a5eef9601ed7c9206/src/lib/components/Canvas/Piece.svelte#L45) so that it can be saved along with other canvas state, but this can be done as a separate issue too and would not be high priority
